internal/tiltfile/k8s.go

```diff
@@ -382,43 +382,52 @@ func (s *tiltfileState) k8sResourceV2(thread *starlark.Thread, fn *starlark.Buil
 		return nil, err
 	}
 
+	resourceName := workload
+	needsToHaveObjects := false
 	if workload == "" {
```

Comment:
uh oh, you don't require anywhere that if no `workload` passed, `new_name` must be passed. e.g. this test I wrote passes and it shouldn't:
```
func TestK8sResouceNonWorkloadRequiresNewName(t *testing.T) {
	f := newFixture(t)
	defer f.TearDown()

	f.yaml("secret.yaml", secret("bar"))
	f.yaml("namespace.yaml", namespace("baz"))

	f.file("Tiltfile", `
k8s_yaml('secret.yaml')
k8s_yaml('namespace.yaml')
k8s_resource(objects=['bar', 'baz:namespace:default'])
`)

	f.load()

	f.assertNextManifest("", k8sObject("bar", "Secret"), k8sObject("baz", "Namespace"))
	f.assertNoMoreManifests()
}
```

Reply:
Good catch! Thanks.